lib/chef-api/resources/base.rb

```diff
@@ -927,7 +927,7 @@ def to_hash
     #
     # @return [String]
     #
-    def to_json
+    def to_json(state = nil)
```

Comment:
Well we could do that (and I can change the PR if you want) but this changes the prototype of method, so I'd rather keep it as one optional argument. but if you insist, I cant change it :-)